src/tests/Common/CLRTest.Jit.targets

```diff
@@ -187,6 +192,115 @@ IF NOT DEFINED DoLink (
+               Condition="'%(Compile.HasDisasmCheck)' == 'true'"
+               CopyToOutputDirectory="PreserveNewest" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GetDisasmCheckBashScript"
+          Returns="$(HasBashDisasmCheck);$(BashDisasmOutputFile);$(BashDisasmCheckPostCommands)"
+          DependsOnTargets="GetDisasmCheckData">
+    <PropertyGroup>
+      <HasBashDisasmCheck>false</HasBashDisasmCheck>
+      <HasBashDisasmCheck Condition="'$(HasDisasmCheck)' == 'true' and '$(RuntimeFlavor)' == 'coreclr' and ('$(TargetOS)' == 'Linux' or '$(TargetOS)' == 'OSX') and ('$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64')">true</HasBashDisasmCheck>
+
+      <BashDisasmOutputFile>/dev/null</BashDisasmOutputFile>
+      <BashDisasmOutputFile Condition="'$(HasBashDisasmCheck)' == 'true'">$(scriptPath)__jit_disasm.out</BashDisasmOutputFile>
+
+      <BashDisasmListOutputFile>/dev/null</BashDisasmListOutputFile>
+      <BashDisasmListOutputFile Condition="'$(HasBashDisasmCheck)' == 'true'">$(scriptPath)__jit_disasm_list.out</BashDisasmListOutputFile>
+
+      <BashCLRTestPreCommands Condition="'$(HasBashDisasmCheck)' == 'true'"><![CDATA[
+@(DisasmCheckFiles -> '  dotnet $CORE_ROOT/SuperFileCheck/SuperFileCheck.dll --csharp-list-method-names "%(Identity)" --allow-unused-prefixes --check-prefixes=CHECK,$(TargetArchitecture.ToUpperInvariant()),$(TargetArchitecture.ToUpperInvariant())-$(TargetOS.ToUpperInvariant()) > "$(BashDisasmListOutputFile)"
+  ERRORLEVEL=$?
+  export COMPlus_JitDisasm=`cat $(BashDisasmListOutputFile)`
+  export COMPlus_JitDiffableDasm=1
+  export COMPlus_JitStdOutFile=$(BashDisasmOutputFile)
+  if [[ $ERRORLEVEL -ne 0 ]]
+  then
+    echo EXECUTION OF FILECHECK - FAILED $ERRORLEVEL
+    exit 1
+  fi', '%0a')
+]]>
+      </BashCLRTestPreCommands>
+
+      <BashDisasmCheckPostCommands></BashDisasmCheckPostCommands>
+      <BashDisasmCheckPostCommands Condition="'$(HasBashDisasmCheck)' == 'true'"><![CDATA[
+if [[ -n $COMPlus_JitDisasm ]]; then
+  @(DisasmCheckFiles -> '  dotnet $CORE_ROOT/SuperFileCheck/SuperFileCheck.dll --csharp "%(Identity)" --allow-unused-prefixes --check-prefixes=CHECK,$(TargetArchitecture.ToUpperInvariant()),$(TargetArchitecture.ToUpperInvariant())-$(TargetOS.ToUpperInvariant()) --dump-input-context 25 --input-file "$(BashDisasmOutputFile)"
+    ERRORLEVEL=$?
+    if [[ $ERRORLEVEL -ne 0 ]]
+    then
+      echo EXECUTION OF FILECHECK - FAILED $ERRORLEVEL
+      exit 1
+    fi', '%0a')
+fi
+]]>
+      </BashDisasmCheckPostCommands>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="GetDisasmCheckBatchScript"
+          Returns="$(HasBatchDisasmCheck);$(BatchDisasmOutputFile);$(BatchDisasmCheckPostCommands)"
+          DependsOnTargets="GetDisasmCheckData">
+    <PropertyGroup>
+      <HasBatchDisasmCheck>false</HasBatchDisasmCheck>
+      <HasBatchDisasmCheck Condition="'$(HasDisasmCheck)' == 'true' and '$(RuntimeFlavor)' == 'coreclr'">true</HasBatchDisasmCheck>
```

Comment:
Nit - in <code>HasBashDisasmCheck</code> you explicitly check that the OS is <code>Linux</code> or <code>OSX</code> but here you're not checking for <code>Windows</code>, that seems somewhat inconsistent. As I assume only one of the two targets gets executed based on the OS, perhaps we could remove the check for Linux / OSX from the <code>HasBashDisasmCheck</code> condition.

Reply:
`Batch` only runs on windows so there isn't a need for a check.

For `Bash`, we need to explicitly check for Linux and OSX as there are other OSes like 'FreeBSD' that do not support the FileCheck tool.

Reply:
I see, thanks for clarifying.

Reply:
By the way, if that is the case, don't we still need to check for x64 / arm64 in the Windows version as you only support this type of checks on 64-bit architectures according to the PR description?

Reply:
For x86 archs, CI stills runs on a 64-bit machine so FileCheck can still work for testing.